spring/src/main/java/reactor/blockhound/spring/BlockHoundRunListener.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2019-Present Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.blockhound.spring;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.SpringApplicationRunListener;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.ConfigurableEnvironment;
+import reactor.BlockHound;
+import reactor.blockhound.integration.BlockHoundIntegration;
+
+/**
+ * This {@link SpringApplicationRunListener} installs the agent when an app is starting
+ * by calling {@link BlockHound#install(BlockHoundIntegration...)}.
+ */
+public class BlockHoundRunListener implements SpringApplicationRunListener, Ordered {
+
+    public BlockHoundRunListener(SpringApplication application, String[] args) {
+
+    }
+
+    @Override
+    public int getOrder() {
+        return Ordered.LOWEST_PRECEDENCE;
```

Comment:
@philwebb is it correct to use `LOWEST` here?
Wanted: "be one of the very first listeners applied"

Reply:
> Wanted: "be one of the very first listeners applied"

@bsideup, you'd then want to use `HIGHEST_PRECEDENCE` which has the lowest numerical value.

`org.springframework.boot.SpringApplication.getSpringFactoriesInstances(Class<T>, Class<?>[], Object...)` sorts the listeners using `org.springframework.core.annotation.AnnotationAwareOrderComparator.sort(List<?>)`.

Reply:
Thank you, Sam! TBH I always make this mistake :(